tests/integration/test_subscriptions_cli.py

```diff
@@ -133,3 +133,114 @@ def test_subscriptions_bad_request(cmd_arg, runner_input):
 
     assert result.exit_code == 2  # bad parameter.
     assert "Request does not contain valid json" in result.output
+
+
+def test_subscriptions_cancel_failure(monkeypatch):
+    """Cancel command exits gracefully from an API error."""
+
+    monkeypatch.setattr(planet.cli.subscriptions, '_fake_subs', {})
+
+    # Begin CLI command.
+    import click
+    from planet.cli.subscriptions import (_cancel_fake_sub,
+                                          coro,
+                                          echo_json,
+                                          translate_exceptions)
+    from planet.exceptions import PlanetError
+
+    @subscriptions.command(name='cancel')
+    @click.argument('subscription_id')
+    @click.option('--pretty', is_flag=True, help='Pretty-print output.')
+    @click.pass_context
+    @translate_exceptions
+    @coro
+    async def cancel_subscription(ctx, subscription_id, pretty):
+        """Cancels a subscription and prints the API response.
+
+        This implementation is only a placeholder. To begin, instead
+        of mocking calls to the Subscriptions API, we'll use a
+        collection of fake subscriptions (the all_subs object).
+        After we refactor we will change to mocking the API.
+
+        """
+        # Begin fake subscriptions service. Note that the Subscriptions
+        # API will report missing keys differently, but the Python API
+        # *will* raise PlanetError like this.
+        try:
+            sub = _cancel_fake_sub(subscription_id)
```

Comment:
Why not import `_fake_subs` from planet.cli.subscriptions and pop the subscription from that? Because we could miss the monkeypatching.